prov/shm/configure.m4

```diff
@@ -14,9 +14,13 @@ AC_DEFUN([FI_SHM_CONFIGURE],[
 	AS_IF([test x"$enable_shm" != x"no"],
 	      [
 	       # check if CMA support are present
-	       AC_CHECK_FUNC([process_vm_readv],
-			     [cma_happy=1],
-			     [cma_happy=0])
+	       shm_ARCH=`uname -m | sed -e 's,\(i[456]86\|athlon$$\),i386,'`
+	       AS_IF([test x$linux = x1 && test x$shm_ARCH = xx86_64],
```

Comment:
Let's move the architecture check into the top configure.ac, so the result is available for all providers.  Search for AC_CANONICAL_HOST and $host_cpu.  This will be similar to the $host_os check that's there now, and avoid trying to parse uname.

Reply:
Actually, you may just need to do something like:
```
AS_IF([test x$linux = x1 && test "$host_cpu" = "x86_64"],
```
host_cpu should already be set.

Reply:
Where does $host_cpu get set?
Also I stole this check from the psm2 configure.ac. Would you like me to change it there too?

Reply:
AC_CANONICAL_HOST sets $host_cpu and $host_os and is in the topmost configure.ac.  Yes, please change the psm2 configure as well.